pkg/agent/flowexporter/connections/l7_listener.go

```diff
@@ -0,0 +1,215 @@
+	"k8s.io/klog/v2"
+
+	"antrea.io/antrea/pkg/agent/controller/l7visibility"
+	"antrea.io/antrea/pkg/agent/flowexporter"
+	k8sutil "antrea.io/antrea/pkg/util/k8s"
+	"antrea.io/antrea/pkg/util/podstore"
+)
+
+const socketPath = "/var/run/suricata/suricata_eve.socket"
+
+type podL7VisibilityAttrGetter interface {
+	IsL7VisibilityRequested(podNN string, ingress bool) bool
+}
+
+// Define struct to hold the L7 flow JSON values
+type Http struct {
+	Hostname      string `json:"hostname"`
+	URL           string `json:"url"`
+	UserAgent     string `json:"http_user_agent"`
+	ContentType   string `json:"http_content_type"`
+	Method        string `json:"http_method"`
+	Protocol      string `json:"protocol"`
+	Status        int32  `json:"status"`
+	ContentLength int32  `json:"length"`
+}
+
+// Event JSON format is defined by Suricata
+// See https://docs.suricata.io/en/latest/output/eve/eve-json-format.html?highlight=HTTP%20event#event-types
+type JsonToEvent struct {
+	Timestamp   string     `json:"timestamp"`
+	FlowID      int64      `json:"flow_id"`
+	InInterface string     `json:"in_iface"`
+	EventType   string     `json:"event_type"`
+	VLAN        []int32    `json:"vlan"`
+	SrcIP       netip.Addr `json:"src_ip"`
+	SrcPort     int32      `json:"src_port"`
+	DestIP      netip.Addr `json:"dest_ip"`
+	DestPort    int32      `json:"dest_port"`
+	Proto       string     `json:"proto"`
+	TxID        int32      `json:"tx_id"`
+	HTTP        *Http      `json:"http"`
+}
+
+type L7Listener struct {
+	l7Events                  map[flowexporter.ConnectionKey]map[string]map[int32]interface{}
+	l7mut                     sync.Mutex
+	suricataEventSocketPath   string
+	podL7VisibilityAttrGetter podL7VisibilityAttrGetter
+	podStore                  podstore.Interface
+}
+
+func NewL7Listener(
+	l7VisibilityController *l7visibility.L7VisibilityController,
+	podstore podstore.Interface,
+	o *flowexporter.FlowExporterOptions) *L7Listener {
+
+	return &L7Listener{
+		l7Events:                  make(map[flowexporter.ConnectionKey]map[string]map[int32]interface{}),
+		suricataEventSocketPath:   socketPath,
+		podL7VisibilityAttrGetter: l7VisibilityController,
+		podStore:                  podstore,
+	}
+}
+
+func (l *L7Listener) Run(stopCh <-chan struct{}) {
+	var listener net.Listener
+	err := wait.PollImmediate(time.Second, time.Second*60, func() (bool, error) {
+		var err error
+		if err = os.Remove(l.suricataEventSocketPath); err != nil && !os.IsNotExist(err) {
+			return false, fmt.Errorf("failed to remove stale socket: %w", err)
+		}
+		if err = os.MkdirAll(filepath.Dir(l.suricataEventSocketPath), 0755); err != nil {
+			return false, fmt.Errorf("failed to create directory %s: %w", l.suricataEventSocketPath, err)
+		}
+		listener, err = net.Listen("unix", l.suricataEventSocketPath)
+		if err != nil {
+			return false, fmt.Errorf("failed to listen on socket: %w", err)
+		}
+		return true, nil
+	})
+	if err != nil {
+		klog.Fatalf("Failed to initialize L7Listener: %v", err)
```

Comment:
I might have commented this before, it's not proper to exit the whole program here. Flow Exporter should not be fatal to antrea-agent, It can keep retrying with a backoff, just make the error visible when user looking at the log, don't make it impact the whole node's network.